### PR TITLE
Updating to htsjdk 1.137

### PR DESCRIPTION
**Pull request by lbergelson**
updating bams, sams, and cram to sam spec version 1.5 (some invalid bams were not updated)
updated interval list headers for bed tests from v 1.4 - 1.5
updating several tests to give a better error message if an index IS present when it's expected to not be

### CODE REVIEW

**Comment by lbergelson** [comment]
for the curious (and my own future reference) I used https://gist.github.com/lbergelson/fd84d30b37fec28eb9da to do the update

File: src/test/java/org/broadinstitute/hellbender/engine/ReadsDataSourceUnitTest.java [hunk 1]
```diff
@@ -49,17 +49,26 @@ public void testHandleNonExistentFile() {
     @Test(expectedExceptions = UserException.class)
     public void testHandleUnindexedFileWithIntervals() {
         // Cannot initialize a reads source with intervals unless all files are indexed
-        ReadsDataSource readsSource = new ReadsDataSource(new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam"));
+        final File unindexed = new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam");
+        assertUnindexed(unindexed);
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/ReadsDataSourceUnitTest.java]: Change this to: `Assert.assertNotNull(SamFiles.findIndex(unindexed), "Expected file to have no index, but found an index file");`

File: src/test/java/org/broadinstitute/hellbender/engine/ReadsDataSourceUnitTest.java [hunk 2]
```diff
@@ -49,17 +49,26 @@ public void testHandleNonExistentFile() {
...
+        ReadsDataSource readsSource = new ReadsDataSource(unindexed);
         readsSource.setIntervalsForTraversal(Arrays.asList(new SimpleInterval("1", 1, 5)));
     }
 
     @Test(expectedExceptions = UserException.class)
     public void testHandleUnindexedFileQuery() {
         // Construction should succeed, since we don't pass in any intervals, but the query should throw.
-        ReadsDataSource readsSource = new ReadsDataSource(new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam"));
+        final File unindexed = new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam");
+        assertUnindexed(unindexed);
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/ReadsDataSourceUnitTest.java]: Change this to `Assert.assertNotNull(SamFiles.findIndex(unindexed), "Expected file to have no index, but found an index file");`

**Comment by droazen** [comment]
Can you confirm (via a script) that all bam/sam files touched in this commit have the same number of records after conversion?

**Comment by lbergelson** [comment]
Will do.

**Comment by droazen** [comment]
Review complete -- back to @lbergelson

**Comment by lbergelson** [comment]
So, I ran the following to count the number of reads in each sam/cram/bam:

```
➜  hellbender git:(master) ✗ zargs -n 4 -- **/*.(b|s|cr)am -- samtools view -cT src/test/resources/org/broadinstitute/hellbender/tools/split_reads.fasta > counts_original
[W::hts_close] EOF marker is absent. The input is probably truncated.
➜  hellbender git:(master) ✗ git checkout lb_update_htsjdk
Switched to branch 'lb_update_htsjdk'
Your branch is up-to-date with 'origin/lb_update_htsjdk'.
➜  hellbender git:(lb_update_htsjdk) ✗ zargs -n 4 -- **/*.(b|s|cr)am -- samtools view -cT src/test/resources/org/broadinstitute/hellbender/tools/split_reads.fasta > counts_updated
[W::hts_close] EOF marker is absent. The input is probably truncated.
➜  hellbender git:(lb_update_htsjdk) ✗ diff counts_updated counts_original
```

[counts_original | uploaded via ZenHub](https://files.zenhub.io/55b7fd0b8bc83ca469fdd8f2)

[counts_updated | uploaded via ZenHub](https://files.zenhub.io/55b7fd0b8bc83ca469fdd8f3)

The counts are identical, so I think we haven't lost any reads.

**Comment by lbergelson** [comment]
@droazen Is that sufficient evidence that the files are similar?

**Pull request merged**